### ISSUE TITLE
Support for 2FA natively in Rancher

### ISSUE THREAD

**Issue by leflambeur** [rancher/rancher#26759]
**What kind of request is this (question/bug/enhancement/feature request):**
Feature Request

**Description**
Can Rancher natively support 2FA solutions specifically including smartcard solutions for govt/healthcare driven workloads where this is a common requirement

**Other details that may be helpful:**
Environment in question uses LDAPS and needs 2FA support on the Rancher side, it is known other SAML driven solutions can do this on the Auth Provider side but it would be more appropriate in many cases to have Rancher working directly to provide 2FA to reduce complexity that results in potential MitM vulns

I can see a similar issue open for local users - #7506

**Comment by chauffer** [comment]
@leflambeur #7506 started as [a request for generic 2FA as well](https://github.com/rancher/rancher/issues/7506#event-2019156704), more than 3 years ago now!

**Comment by lptarik** [comment]
When will have this feature?

**Comment by CyberCyclone** [comment]
While using another Auth Provider allows users to make sue of their MFA, since Rancher keeps local users enabled ('Local Authentication is always enabled, but you may select another additional authentication provider from those shown below.'), local MFA is essential. I don't see how businesses (or users in general) can use Rancher without this basic security need.

**Comment by sim1** [comment]
#7506 has been open for over 5 years now. I wouldn't expect this to be tackled anytime soon...

**Comment by michaelhofer** [comment]
I agree, having native 2FA support is critical, since the local users cannot be disabled.

**Comment by xhejtman** [comment]
+1

**Comment by rosinec** [comment]
Would be really helpfull if the Rancher could check for acr/amr claims in the token requiring and confirming an existence of MFA method used on an upstream OIDC IDP.
